app/javascript/app/ssn-field.js

```diff
@@ -48,8 +48,20 @@ function formatSSNField() {
 
       sync();
       toggle.addEventListener('change', sync);
+
+      const ssnBox = document.getElementById('doc_auth_ssn');
+      const maxLength = 9;
+
+      function limitLength() {
+        if (this.value.length > maxLength) {
+          this.value = this.value.slice(0, maxLength);
+        }
+      }
+
+      ssnBox.addEventListener('keyup', limitLength.bind(ssnBox));
+      ssnBox.addEventListener('keydown', limitLength.bind(ssnBox));
```

Comment:
We should use `input` event if we're trying to capture all changes to the input text. These won't account for a case where a user right-clicks and pastes text.

```suggestion
      ssnBox.addEventListener('input', limitLength.bind(ssnBox));
```

Reply:
nice catch!